modin/core/storage_formats/pandas/parsers.py

```diff
@@ -312,7 +312,12 @@ def parse(chunks, **kwargs):
         for fname, start, end in chunks:
             if start is not None and end is not None:
                 # pop "compression" from kwargs because bio is uncompressed
-                with OpenFile(fname, "rb", kwargs.pop("compression", "infer")) as bio:
+                with OpenFile(
+                    fname,
+                    "rb",
+                    kwargs.pop("compression", "infer"),
+                    **(kwargs.pop("storage_options", None) or {}),
```

Comment:
will that work?
```suggestion
                    **kwargs.pop("storage_options", {}),
```

Reply:
btw, why do we remove storage options from the kwargs here? this `OpenFile` happens in a for-loop, so if we pop a `storage format` at the first iteration, all the following chunks would get an empty parameter for the `OpenFile`, was it done on purpose?

Reply:
No, `storage_options` value in kwargs can be None. In this case we'll get an error on unpacking (** NoneType).

Reply:
> btw, why do we remove storage options from the kwargs here? this `OpenFile` happens in a for-loop, so if we pop a `storage format` at the first iteration, all the following chunks would get an empty parameter for the `OpenFile`, was it done on purpose?

`storage_options` and `compression` is removed from kwargs because `read_csv` from `BytesIO` is used (those parameters isn't supported by it).

It was done by mistake.